---
title: C28198
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 78988ce7995e3ae5fde0a6391cc0cf7b46128362
ms.sourcegitcommit: 485ffaedb1ade71490f11cf05962add1718945cc
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 10/16/2019
ms.locfileid: "72448302"
---
# <a name="c28198"></a>C28198
Upozornění C28198: pravděpodobně nevrácení paměti z důvodu výjimky.

 Toto upozornění znamená, že přidělená paměť není uvolněna po vyvolání výjimky. Příkaz na konci cesty může vyvolat výjimku. Paměť byla předána funkci, která mohla uložit kopii pro pozdější uvolnění.

 Toto upozornění je velmi podobné upozornění [C28197](../code-quality/c28197.md). Poznámky, které jsou doporučeny pro použití s upozorněním [C28197](../code-quality/c28197.md) , lze také použít zde.

## <a name="example"></a>Příklad
 Následující příklad kódu vygeneruje toto upozornění:

```
char *p1 = new char[10];
char *p2 = new char[10];

test(p1);   // does not save a copy of p

delete[] p2;
delete[] p1;
```

 Následující příklad kódu předchází tomuto upozornění:

```
char *p1 = new char[10];
char *p2 = NULL;

test(p1);   // does not save a copy of p
try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
```
